frappe/public/js/frappe/model/model.js

```diff
@@ -650,6 +650,27 @@ $.extend(frappe.model, {
 		}
 	},
 
+	round_floats_childs_in: function(arr, fieldnames) {
+		var cached_precision = [];
+		$.each(arr || [], function(k, item) {
+			if(!fieldnames) {
+				fieldnames = frappe.meta.get_fieldnames(item.doctype, item.parent,
+					{"fieldtype": ["in", ["Currency", "Float"]]});
+			}
+
+			for(var i=0, j=fieldnames.length; i < j; i++) {
+				var fieldname = fieldnames[i];
+				
+				if (k == 0) { // cache precision on first arr position, the precision will be the same for all rows.
```

Comment:
instead of doing this in the loop and checking if the index is 0, you can move this part out and first initialize all precision values.

Reply:
done